### ISSUE TITLE
B1QUAD/FDLog_Enhanced_python3

### ISSUE THREAD

**Issue by scotthibbs** [B1QUAD/B1QUAD#1]
David, 

I was wondering if you still had this saved somewhere... Seems the project isn't on your github anymore. Unfortunately, I didn't get a copy for myself. 

I'm a very amateur programmer and would like to switch this project to python 3. Sounded like you might have a pretty stable version and just needed some testing on the networking section. 

My email is scotthibbs at g mail. 
Thanks
Scott

**Comment by B1QUAD** [comment]
Hi Mr. Hibbs, I am sorry it's taken so long to reply to your github issue, I had a final exam for a summer physics class yesterday and am just now catching up on everything else. Just in case anyone else wants a copy of the source code I decided to publish it here.

The software seems to be working except for an exception that is thrown because of what looks like some sort of type mismatch error at first glance (there could be a deeper root cause but the memories of debugging are a bit hazy a year later). 

```
File "/home/.../FDLog_Enhanced_python3/./FDLog_SCICSG.py", line 1281, in ckauth
    h, m = msg.split('\n', 1).encode('utf-8')
TypeError: a bytes-like object is required, not 'str'
```
Hopefully this is the only remaining bug and you will be able to sort it out. I am also an amateur python programmer and ended up just switching to paper logs for that year's field day. However, if you do get a stable python3 version running I would love to try it out!

Best of luck and have a wonderful day,
-David

[FDLog_Enhanced_python3.zip](https://github.com/B1QUAD/B1QUAD/files/9075537/FDLog_Enhanced_python3.zip)

**Comment by scotthibbs** [comment]
David,

THANK YOU THANK YOU THANK YOU!!!

That made my day.

I have spent all week adding copy/paste functionality to FDLog_Enhanced
simply because Alan Biocca (the program's original author) had a note in
the program notes where someone asked for it probably in the 90's. I can't
see why anyone would need it now, but I'm having fun adding it.

After I get this "enhancement" finished (paste now works but I broke the
copy function doing it), I fully intend to match our programs together and
transition this project to python 3. I couldn't imagine doing this from
scratch since I haven't even looked at python 3 yet.

I really appreciate you sending it to me. I'll do my best to get it
working. Best of luck to you in your college studies,

73
KD4SIR
Scott Hibbs
